ndcube/ndcube.py

```diff
@@ -952,3 +962,234 @@ def to(self, new_unit, **kwargs):
+            True elements cause corresponding uncertainty elements to be ignored.
+            False elements cause corresponding uncertainty elements to be propagated.
+            Must have same shape as above.
+            If None, no uncertainties are ignored.
+
+        All kwarg inputs to the rebin method are also passed on transparently to the
+        propagation function. Hence additional inputs to the propagation function can be
+        included as kwargs to `ndcube.NDCube.rebin`.
+
+        The shape of the uncertainty, data and mask inputs are such that the first
+        dimension represents the pixels in a given bin whose data and uncertainties
+        are aggregated by the rebin process.  The shape of the remaining dimensions
+        must be the same as the final rebinned data.  A silly but informative
+        example of a custom propagation function might be::
+
+             def my_propagate(uncertainty, data, mask, **kwargs):
+                 # As a silly example, propagate uncertainties by summing those in same bin.
+                 # Note not all args are used, but function must accept them.
+                 n_pixels_per_bin = data.shape[0]  # 1st dimension of inputs gives pixels in bin.
+                 final_shape = data.shape[1:]  # Trailing dims give shape of put rebinned data.
+                 # Propagate uncerts by adding them.
+                 new_uncert = numpy.zeros(final_shape)
+                 for i in range(n_pixels_per_bin):
+                     new_uncert += uncertainty.array[i]
+                 # Alternatively: new_uncerts = uncertainty.array.sum(axis=0)
+                 return type(uncertainty)(new_uncert)  # Convert to original uncert type and return.
+        """
+        # Sanitize input.
+        if new_unit is None:
+            new_unit = self.unit
+        # Make sure the input bin dimensions are integers.
+        bin_shape = np.rint(bin_shape).astype(int)
+        offsets = (bin_shape - 1) / 2
+        if all(bin_shape == 1):
+            return self
+        # Ensure bin_size has right number of entries and each entry is an
+        # integer fraction of the array shape in each dimension.
+        data_shape = self.dimensions.value.astype(int)
+        naxes = len(data_shape)
+        if len(bin_shape) != naxes:
+            raise ValueError("bin_shape must have an entry for each array axis.")
+        if (np.mod(data_shape, bin_shape) != 0).any():
+            raise ValueError(
+                "bin shape must be an integer fraction of the data shape in each dimension. "
+                f"data shape: {data_shape};  bin shape: {bin_shape}")
+
+        # Reshape array so odd dimensions represent pixels to be binned
+        # then apply function over those axes.
+        m = None if (self.mask is None or self.mask is False or use_masked_values) else self.mask
+        data = self.data
+        if m is None:
+            data = self.data
+        else:
```

Comment:
With the line above this is equivalent?
```suggestion
        if m is not None:
```